core/shared/crypto.ts

```diff
@@ -0,0 +1,37 @@
+import { verify } from 'bitcoinjs-message';
+import { MainNet } from '@defichain/jellyfish-network';
+
+export class Crypto {
+  static verifySignature(message: string, address: string, signature: string): boolean {
+    let isValid = false;
+    try {
+      isValid = this.verify(message, address, signature);
+    } catch (e) {}
+
+    if (!isValid) {
+      isValid = this.fallbackVerify(message, address, signature);
+    }
+    return isValid;
+  }
+
+  private static fallbackVerify(message: string, address: string, signature: string) {
+    let isValid = false;
+    const flags = [...Array(12).keys()].map((i) => i + 31);
+    for (const flag of flags) {
+      const flagByte = Buffer.alloc(1);
+      flagByte.writeInt8(flag);
+      let sigBuffer = Buffer.from(signature, 'base64').slice(1);
+      sigBuffer = Buffer.concat([flagByte, sigBuffer]);
+      const candidateSig = sigBuffer.toString('base64');
+      try {
+        isValid = this.verify(message, address, candidateSig);
+        if (isValid) break;
+      } catch (e) {}
+    }
+    return isValid;
+  }
+
+  private static verify(message: string, address: string, signature: string): boolean {
+    return verify(message, address, signature, MainNet.messagePrefix);
```

Comment:
Shouldn't we decide based on environment, which network we are using? At least API is running on TestNet for loc and dev

Reply:
Yes, would be right. I was just too lazy, because the prefix is the same for all networks.